spec/rubocop/cop/bundler/ordered_gems_spec.rb

```diff
@@ -155,6 +155,17 @@
           gem 'rubocop'
         RUBY
       end
+
+      it 'does not register an offense' do
+        expect_no_offenses(<<~RUBY)
```

Comment:
My understanding is that it should `expect_offense` and correct this to 
```
          # test
          gem 'rspec'
          
          # For code quality
          gem 'rubocop'
          # For
```

Reply:
@marcandre The original code already meets your needs. You can copy following tests to `ordered_gems_spec.rb` and try it.
```
  context 'with TreatCommentsAsGroupSeparators: false' do
      it 'registers an offense' do
        expect_offense(<<~RUBY)
        gem 'rubocop',
            '0.1.1'
        # For

        # test
        gem 'rspec'
        ^^^^^^^^^^^ Gems should be sorted in an alphabetical order within their section of the Gemfile. Gem `rspec` should appear before `rubocop`.
        RUBY

        expect_correction(<<~RUBY)
        # For

        # test
        gem 'rspec'
        gem 'rubocop',
            '0.1.1'
        RUBY
      end
    end
```

Reply:
```ruby
    it 'registers an offense' do
        expect_offense(<<~RUBY)
          # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
          gem 'rails', '6.1.1'

          # Environment variable support, needs to be loaded before anything
          gem 'dotenv-rails', require: 'dotenv/rails-now'

          # Used to communicate with Braintree, our payment gateway provider
          gem 'braintree'
        RUBY

        expect_correction(<<~RUBY)
          # Used to communicate with Braintree, our payment gateway provider
          gem 'braintree'

          # Environment variable support, needs to be loaded before anything
          gem 'dotenv-rails', require: 'dotenv/rails-now'

          # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
          gem 'rails', '6.1.1'
        RUBY
      end
```

Reply:
(copy-pasted from the original issue); spec does not pass.

Reply:
In my understanding, no matter ` TreatCommentsAsGroupSeparators` true or not,  blank lines are always group separator.  So I think the following codes are good. 
```
# Good

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '6.1.1'

# Environment variable support, needs to be loaded before anything
 gem 'dotenv-rails', require: 'dotenv/rails-now'
```

BTW, I suspected you are  misunderstood what @coding-bunny said. He posted following codes:
```
# Used to communicate with Braintree, our payment gateway provider
gem 'braintree'
# Used to speed up development boot time
gem 'bootsnap', '>= 1.4.2', require: false
```
and said 
> I expect the Bundler/OrderedGems to report the following: Bootsnap should be before braintree
So I think that he forget to set TreatCommentsAsGroupSeparators as false, which is not a bug at all.

Reply:
Blank likes should always be group separaters, otherwise it becomes impossible to have gems such as `dotenv-rails` to be at the top of a Gemfile and use their functionality to load `ENV` variables for following gems.

I'm not familiar with the `TreatCommentsAsGroupSeparators`, I can actually check that if that was missed in the relevant project to see if that produces the desired behavior. The goal was to keep gems in alphabetical order when they're all together without linebreaks, but also adhere to the Cop that gems with a version need to have documentation above them

Reply:
okay I've verified the gem : 

Setting the `TreatCommentsAsGroupSeparators` to `false` actually does what I want:

```
gem 'oauth2'
gem 'paper_trail'
# Webserver
gem 'puma', '~> 5.0'
gem 'redis-activesupport'
gem 'rollbar'
# Sass for stylesheet compilation
gem 'sass-rails', '>= 6'
# Rails turbolinks
gem 'turbolinks', '~> 5'
# Webpacker to compile our JS
gem 'webpacker', '~> 5.0'
gem 'webpacker-routes'
```

This is considered valid using both Cops, and is good.
So....euh I'm sorry for the unneeded work xD

Reply:
Oh, I'm sorry, I did misunderstand, and I didn't think blank lines were always meant as separators.

Reply:
So in what scenario will blank lines not be used as separators？
Should this issue and pr be closed?

Reply:
I'm afraid it should always be a separator, then. My fault for not understanding that, sorry.